subcerts.go

```diff
@@ -141,67 +148,71 @@ func (cred *credential) marshal() ([]byte, error) {
 	}
 
 	// Write the length of the public_key field.
-	binary.BigEndian.PutUint16(serialized[4:], uint16(len(serializedPublicKey)))
+	binary.BigEndian.PutUint16(serialized[paramsLen:], uint16(len(serializedPublicKey)))
 
-	// Write the public key.
-	return append(serialized, serializedPublicKey...), nil
+	// Write the public_key field.
+	serialized = append(serialized, serializedPublicKey...)
+	cred.raw = serialized
+	return serialized, nil
 }
 
 // unmarshalCredential decodes a credential and returns it.
 func unmarshalCredential(serialized []byte) (*credential, error) {
-	// Bytes 0-3 are the validity time field; bytes 4-6 are the length of the
-	// serialized SubjectPublicKeyInfo.
-	if len(serialized) < 6 {
+	paramsLen := 8
```

Comment:
Is there a reason why paramsLen := 8  not 10?

If there is - comment why you do +2. If not make it 10

Reply:
Done. The reason is that the parameters are 8 bytes; the +2 is for the length of the public key.